haddock-api/src/Haddock/Convert.hs

```diff
@@ -756,7 +771,7 @@ synifyInstHead (vs, preds, cls, types) = specializeInstHead $ InstHead
     ts' = map (synifyType WithinType vs) ts
     annot_ts = zipWith3 annotHsType is_poly_tvs ts ts'
     is_poly_tvs = mkIsPolyTvs (tyConVisibleTyVars cls_tycon)
-    synifyClsIdSig = synifyIdSig DeleteTopLevelQuantification vs
+    synifyClsIdSig = synifyIdSig True DeleteTopLevelQuantification vs
```

Comment:
Why do you hardcode `PrintRuntimeReps` to `True` here instead of consulting the flag?

Reply:
Yeah. So this brings up a bigger question: should we thread through the `RuntimeRep` defaulting logic to everything involving `Type` in this module? Perhaps so, but that is going to be a fair bit of plumbing.

I think it is simpler to start with defaulting only top-level signatures, but I'm willing to be told otherwise.

Reply:
If your goal is to avoid having any `RuntimeRep` quantification unless explicitly enabled, I think you'll eventually need to tackle class instances as well. After all, you can define levity polymorphic instances like so:

```haskell
λ> class C a
λ> instance forall (r :: RuntimeRep) a (b :: TYPE r). C (a -> b)
λ> :i C
class C (a :: k)        -- Defined at <interactive>:2:1
instance [safe] forall a b. C (a -> b)
  -- Defined at <interactive>:3:10
λ> :set -fprint-explicit-runtime-reps 
λ> :i C
class C (a :: k)        -- Defined at <interactive>:2:1
instance [safe] forall (r :: RuntimeRep) a (b :: TYPE r).
                C (a -> b)
  -- Defined at <interactive>:3:10
```

Reply:
Gah. Alright. Thanks for the motivating example. 😄

Reply:
Upon further inspection: I can't convince Haddock to ever print out runtime reps for instances. I'm inclined to leave this be for the moment. Plus this part of the `wip/hi-haddock` branch, so I'm trying to stay away from far-reaching changes.